jerry-core/ecma/base/ecma-gc.c

```diff
@@ -1910,9 +1910,11 @@ ecma_gc_free_object (ecma_object_t *object_p) /**< object to free */
             /* Call external free callback if any. */
             ecma_arraybuffer_external_info *array_p = (ecma_arraybuffer_external_info *) ext_object_p;
 
-            if (array_p->free_cb != NULL)
+            jerry_external_arraybuffer_free_callback_t free_cb = JERRY_CONTEXT (external_arraybuffer_free_callback_p);
+
+            if (array_p->user_p != NULL)
```

Comment:
Incorrect check, the `free_cb` must be checked (just as before) otherwise we could call the free_cb if it is NULL.